libs/grok/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -163,48 +164,53 @@ public String groupMatch(String name, Region region, String pattern) {
      * @return named regex expression
      */
     public String toRegex(String grokPattern) {
-        byte[] grokPatternBytes = grokPattern.getBytes(StandardCharsets.UTF_8);
-        Matcher matcher = GROK_PATTERN_REGEX.matcher(grokPatternBytes);
+        StringBuilder res = new StringBuilder();
+        int i = 0;
+        while (true) {
+            if (i++ >= MAX_TO_REGEX_ITERATIONS) {
+                throw new IllegalArgumentException("Can not convert grok patterns to regular expression");
+            }
+            byte[] grokPatternBytes = grokPattern.getBytes(StandardCharsets.UTF_8);
+            Matcher matcher = GROK_PATTERN_REGEX.matcher(grokPatternBytes);
+
+            int result;
+            try {
+                matcherWatchdog.register(matcher);
+                result = matcher.search(0, grokPatternBytes.length, Option.NONE);
+            } finally {
+                matcherWatchdog.unregister(matcher);
+            }
+
+            if (result < 0) {
+                return res.append(grokPattern).toString();
+            }
 
-        int result;
-        try {
-            matcherWatchdog.register(matcher);
-            result = matcher.search(0, grokPatternBytes.length, Option.NONE);
-        } finally {
-            matcherWatchdog.unregister(matcher);
-        }
-        if (result >= 0) {
             Region region = matcher.getEagerRegion();
             String namedPatternRef = groupMatch(NAME_GROUP, region, grokPattern);
-            String subName = groupMatch(SUBNAME_GROUP, region, grokPattern);
-            // TODO(tal): Support definitions
+            String subName = groupMatch(SUBNAME_GROUP, region, grokPattern);// TODO(tal): Support definitions
```

Comment:
This `TODO` should be moved to new line like it was before (sorry, this was unintentional change in my comment)